controllers/idler/idler_controller.go

```diff
@@ -198,6 +184,58 @@ func (r *Reconciler) ensureIdling(ctx context.Context, idler *toolchainv1alpha1.
 	return r.updateStatusPods(ctx, idler, newStatusPods)
 }
 
+// Check if the pod belongs to a controller (Deployment, DeploymentConfig, etc) and scale it down to zero.
+// if it is a standalone pod, delete it.
+// Send notification if the deleted pod was managed by a controller, was a standalone pod that was not completed or was crashlooping
+func deletePodsAndCreateNotification(ctx context.Context, podLogger logr.Logger, pod corev1.Pod, isCrashlooping bool, r *Reconciler, idler *toolchainv1alpha1.Idler) error {
+	lctx := log.IntoContext(ctx, podLogger)
+	logger := log.FromContext(ctx)
+	var podreason string
+	podCondition := pod.Status.Conditions
+	for _, podCond := range podCondition {
+		if podCond.Type == "Ready" {
+			podreason = podCond.Reason
+		}
+	}
+	appType, appName, deletedByController, err := r.scaleControllerToZero(lctx, pod.ObjectMeta)
+	if err != nil {
+		return err
+	}
+	if !deletedByController { // Pod not managed by a controller. We can just delete the pod.
+		log.FromContext(ctx).Info("Deleting pod without controller")
+		if err := r.AllNamespacesClient.Delete(ctx, &pod); err != nil {
+			return err
+		}
+		podLogger.Info("Pod deleted")
+	}
+	if appName == "" {
+		appName = pod.Name
+		appType = "Pod"
+	}
+
+	// If a build pod is in "PodCompleted" status then it was not running so there's no reason to send an idler notification
+	if podreason != "PodCompleted" || deletedByController || isCrashlooping {
```

Comment:
Do we really need the `isCrashlooping` boolean? There's no special treatment right?

Reply:
you're right, we can remove the boolean

Reply:
done in [a147e63](https://github.com/codeready-toolchain/member-operator/pull/630/commits/a147e636496395427a349188b8cd8a91bfc020ea)